dowhy/causal_estimator.py

```diff
@@ -679,6 +685,91 @@ def signif_results_tostr(self, signif_results):
         return s
 
 
+def estimate_effect(
+    treatment: Union[str, List[str]],
+    outcome: Union[str, List[str]],
+    identified_estimand: IdentifiedEstimand,
+    identifier_name: str,
+    method: CausalEstimator,
+    control_value: int = 0,
+    treatment_value: int = 1,
+    test_significance: Optional[bool] = None,
+    evaluate_effect_strength: bool = False,
+    confidence_intervals: bool = False,
+    target_units: str = "ate",
+    effect_modifiers: List[str] = [],
+    fit_estimator: bool = True,
+    method_params: Optional[Dict] = None,
+):
+    """Estimate the identified causal effect.
+
+    Currently requires an explicit method name to be specified. Method names follow the convention of identification method followed by the specific estimation method: "[backdoor/iv].estimation_method_name". Following methods are supported.
+        * Propensity Score Matching: "backdoor.propensity_score_matching"
+        * Propensity Score Stratification: "backdoor.propensity_score_stratification"
+        * Propensity Score-based Inverse Weighting: "backdoor.propensity_score_weighting"
+        * Linear Regression: "backdoor.linear_regression"
+        * Generalized Linear Models (e.g., logistic regression): "backdoor.generalized_linear_model"
+        * Instrumental Variables: "iv.instrumental_variable"
+        * Regression Discontinuity: "iv.regression_discontinuity"
+
+    In addition, you can directly call any of the EconML estimation methods. The convention is "backdoor.econml.path-to-estimator-class". For example, for the double machine learning estimator ("DML" class) that is located inside "dml" module of EconML, you can use the method name, "backdoor.econml.dml.DML". CausalML estimators can also be called. See `this demo notebook <https://py-why.github.io/dowhy/example_notebooks/dowhy-conditional-treatment-effects.html>`_.
+
+
```

Comment:
treatment, outcome need to be added to docstring.
identifier_name to be added to docstring.